### PR TITLE
Convert the method parameter to lowercase

### PR DESCRIPTION
**Pull request by luciy**
In some cases, passing Request Method in uppercase causes some errors.

Example:

```
axios.defaults.headers.post['Content-Type'] = 'application/x-www-form-urlencoded';

return axios({
  method: 'POST',
  url: url
});
```

### CODE REVIEW

**Comment by rubennorte** [comment]
Nice feature. Could you please add a test for it?

**Comment by luciy** [comment]
add test

File: test/specs/requests.spec.js [hunk 1]
```diff
@@ -17,6 +17,22 @@ describe('requests', function () {
     });
   });
 
+  it('should treat method value as lowercase string', function (done) {
+    axios({
+      url: '/foo',
+      method: 'POST'
+    }).then(function (request) {
```
Comment (rubennorte) [inline test/specs/requests.spec.js]: This parameter should be `response`

**Comment by rubennorte** [comment]
Please don't commit changes to the files in `dist`. Could you undo them in the PR?

**Comment by luciy** [comment]
Resubmit the PR at https://github.com/mzabriskie/axios/pull/930

**Pull request closed without merging**